components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java

```diff
@@ -395,7 +395,10 @@ public KeyManagerConfigurationDTO getKeyManagerConfigurationById(String organiza
                 APIConstants.KeyManager.DEFAULT_KEY_MANAGER.equals(keyManagerConfigurationDTO.getName())) {
             APIUtil.getAndSetDefaultKeyManagerConfiguration(keyManagerConfigurationDTO);
         }
-        maskValues(keyManagerConfigurationDTO);
+        if (!KeyManagerConfiguration.TokenType.valueOf(keyManagerConfigurationDTO.getTokenType().toUpperCase())
```

Comment:
same here

Reply:
Here the equal comparison happens between two enums, not strings.